shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,76 @@
+import json
+import math
+import nibabel as nib
+import numpy as np
+import os
+
+
+def read_nii(nii_path, return_image=False):
+    """ Reads a nifti file and returns the corresponding image and info. Also returns the associated json data.
+    Args:
+        nii_path (str): direct path to the .nii or .nii.gz file that is going to be read
+        return_image (:obj:`bool`, optional): Defines if the function should (True) or should not (False) return `image`
+    Returns:
+        info (Nifti1Image): Objet containing various data about the nifti file (returned by nibabel.load)
+        json_data (dict): Contains the different fields present in the json file corresponding to the nifti file
+        image (ndarray): Image contained in the read nifti file. Siemens phase images are rescaled between 0 and 2pi.
+    """
+
+    info = nib.load(nii_path)
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    json_path = nii_path.split('.nii')[0] + '.json'
+
+    if os.path.isfile(json_path):
+        json_data = json.load(open(json_path))
+    else:
+        raise ValueError('Missing json file')
+
+    if return_image:
+        image = np.asanyarray(info.dataobj)
+        # NOTE: nib.load automatically scales the nifti and replaces scl_inter and scl_slope with 'nan' in the header
+        # More info in the "Data scaling" section in https://nipy.org/nibabel/nifti_images.html
+        if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens') \
+                and (image_type(json_data) == 'phase'):
+            PHASE_SCALING_SIEMENS = 4096
+            image = image * (2 * math.pi / PHASE_SCALING_SIEMENS)
+
+        return info, json_data, image
+    else:
+        return info, json_data
```

Comment:
Don't think it is a good idea to have multiple ways to return data (2 output versus 3 output). It can confuse users as to what they are getting. I'd say that you should return the image anyway and it will be the users job to ignore it (in Python using _) if they don't need it.

Reply:
So If I don't want to do any calculation on the image should I just return an empty object that the user will ignore ?